pkgs/build-support/fetchmavenartifact/default.nix

```diff
@@ -0,0 +1,61 @@
+# Adaptation of the MIT-licensed work on `sbt2nix` done by Charles O'Farrell
+
+{ fetchurl, stdenv }:
+let
+  defaultRepos = [
+    http://central.maven.org/maven2
+    http://oss.sonatype.org/content/repositories/releases
+    http://oss.sonatype.org/content/repositories/public
+    http://repo.typesafe.com/typesafe/releases
+  ];
+in
+
+args@
+{ # Example: "org.apache.httpcomponents"
+  groupId
+, # Example: "httpclient"
+  artifactId
+, # Example: "4.3.6"
+  version
+, # Maven repository from where to fetch the artifact (example:
+  # http://oss.sonatype.org/content/repositories/public)
+  repo ? null
+, # List of maven repositories from where to fetch the artifact.
+  # Has priority over `repo`.
```

Comment:
That's unfortunate, as now you must override both `repo` and `repos`. And the following expression doesn't work and may cause a happy debugging:
```nix
fetchMavenArtifact {
  groupId = "...";
  artifactId = "...";
  version = "...";
  repo = "...";
}
```

I propose to either make a `repo` a higher priority or drop this parameter.
```nix
fetchMavenArtifact {
  groupId = "...";
  artifactId = "...";
  version = "...";
  repos = [ "..." ];
}
```